test/EzMap.IntegrationTest/AnthenticationControllerTest.cs

```diff
@@ -63,4 +81,33 @@ public async Task SignUp_ProvideCorrectData_UserRecordCreated()
         // successful message returned
         Assert.True(BCrypt.Net.BCrypt.Verify(user.Password, dbUser.Password));
     }
+
+    [Fact]
+    public async Task SignUp_InvalidPasswordProvided_BadRequestAndErrorMessageReturn()
+    {
+        var app = new TestWebAppFactory<Program>();
+        var client = app.CreateClient();
+        using var scope = app.Services.CreateScope();
+        var userSignUpDto = new UserCreationDto()
+        {
+            UserName = "bytetum", DisplayName = "hoangml", Email = "notdeciced@gmail.com", Password = "Aaaaaaa!",
+        };
+
+        var response = await client.PostAsJsonAsync("api/user/SignUp", userSignUpDto);
+
+        var responseString = await response.Content.ReadAsStringAsync();
+
+        var errorObject = JsonSerializer.Deserialize<ProblemDetails>(responseString, new JsonSerializerOptions()
+        {
+            PropertyNameCaseInsensitive = true
+        });
+
+        var comparingProperty = nameof(UserSignInDto.Password);
+
+        var returnedErrorMessage = errorObject.Errors[comparingProperty][0];
+
+        var expectedErrorMessage = "Your password must contain at least one number.";
+
+        Assert.True(expectedErrorMessage.Contains(returnedErrorMessage, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
This doesn't prove that we actually validate upper case, lower case, special characters.

This just proves that "Aaaaaaa!" passes the test.

Please make it so that if I changes the validation logic, the test fails, in that case there is value in the test that it notifies (aka fails) when you fail to meet the contract in the test.